lib/uefi/boot_service_provider.cpp

```diff
@@ -692,7 +692,7 @@ EfiStatus locate_handle_buffer(EfiLocateHandleSearchType search_type,
     }
     return SUCCESS;
   }
-  printf("%s(0x%x, (0x%x 0x%x 0x%x 0x%llx), search_key=0x%lx)\n", __FUNCTION__,
+  printf("%s(0x%x, (0x%x 0x%x 0x%x 0x%llx), search_key=%p)\n", __FUNCTION__,
```

Comment:
Different compilers seem to have different implementation for %p, some include 0x in the front some don't. Is there a way to enforce a consistent formatting for pointers?

Reply:
It's not implemented in the compiler actually. In LK's case it's implemented in the libc. The spec says that it will put a 0x in front of it if it's not zero, but if it's zero it just shows that. Kinda weird, but I get used to it.

Similar to %#x (instead of 0x%x). The #x is the nicer, shorter way of doing it, but it only puts a 0x if it's nonzero.

Reply:
ack